platform/local/dnsmasq.go

```diff
@@ -108,6 +112,70 @@ func newInterface(s byte, i uint16) *Interface {
 	}
 }
 
+// configureNAT creates and append a NAT rule
+// using the interface i as output
+func configureNAT(i string) error {
+	table, err := iptables.New()
+	if err != nil {
+		return fmt.Errorf("unable to get iptables: %w", err)
+	}
+
+	if err := table.AppendUnique("nat", "POSTROUTING", "-j", "MASQUERADE", "-o", i); err != nil {
+		return fmt.Errorf("unable to append rule: %w", err)
+	}
+
+	return nil
+}
+
+// generateVethPair creates and returns a map holding
+// a pair of veth based on a seed
+// where:
+// [0] is the network namespace veth
+// [1] is the host namespace veth
+// [2] is the next hop for adding route
+func generateVethPair(seed int) []string {
+	// we basically use each single bit of the seed
+	// to generate an unique IP range
+	x := 0x10 | (seed & 0x0F)
+	y := ((seed >> 4 & 0x0F) << 4) | (seed >> 8 & 0x0F)
+	hostZ := ((seed >> 12 & 0x0F) << 4) | 0x00
+	nsZ := ((seed >> 12 & 0x0F) << 4) | 0x01
```

Comment:
I get confused^^, maybe just a single shift and then masks:
```suggestion
	start := 0b10101100000100000000000000000000 // 172.16.0.0/12
	host := start | (seed <<1) | 0
	ns := host | 1
	x := host & 0xFF0000
	y := host & 0xFF00
	hostZ := host & 0xFF
	nsZ := ns & 0xFF
```